### PR TITLE
lint: enable stylecheck

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Follow-up to: https://github.com/osmosis-labs/osmosis/issues/1897

Linter that enforces style rules. A subset of rules from the staticcheck binary.

## Brief Changelog

- disable annoying check requiring "Id" to be "ID"
- ensure that receivers are named consistently
- remove duplicate imports that had different identifiers
- remove redundant types in var declarations
- errors messages are not capitalized
- godoc starts with var name

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

**Comment by p0mvn** [comment]
Anecdotally, it feels that this linter caused the largest increase in time it takes to run `golangci-lint run`

However, it does seem that it was helpful in enforcing consistency so might still be worth trying it.

I think it should be another candidate to be potentially removed if linter times become a problem. Instead, we can run it every now and then to clean up the codebase

**Comment by p0mvn** [comment]
> Anecdotally, it feels that this linter caused the largest increase in time it takes to run `golangci-lint run`
> 
> However, it does seem that it was helpful in enforcing consistency so might still be worth trying it.
> 
> I think it should be another candidate to be potentially removed if linter times become a problem. Instead, we can run it every now and then to clean up the codebase

On another look, seems I was wrong - the difference was likely due to cache. After timing these again:
```
git checkout main
time golangci-lint run

real    0m1.441s
user    0m5.352s
sys     0m1.170s
```
```
git checkout roman/lint-stylecheck
time golangci-lint run

real    0m1.465s
user    0m5.423s
sys     0m1.231s
```

Going to leave this here still in case anyone has thoughts

**Review by alexanderbez — APPROVED** [review]

File: x/gamm/pool-models/stableswap/pool.go [hunk 1]
```diff
@@ -196,59 +196,59 @@ func (pa Pool) CalcInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coins, tokenInDen
 	return sdk.NewCoin(tokenInDenom, tokenInAmt), nil
 }
 
-func (pa *Pool) SwapInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coins, tokenInDenom string, swapFee sdk.Dec) (tokenIn sdk.Coin, err error) {
-	tokenIn, err = pa.CalcInAmtGivenOut(ctx, tokenOut, tokenInDenom, swapFee)
+func (p *Pool) SwapInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coins, tokenInDenom string, swapFee sdk.Dec) (tokenIn sdk.Coin, err error) {
```
Comment (mattverse) [inline x/gamm/pool-models/stableswap/pool.go:199]: yay pa -> p !

**Review by mattverse — APPROVED** [review]

**Comment by mattverse** [comment]
doesn't seem like a big increase in time / a big overhead just from the stats you provided?

**Pull request merged**